qa/smoke-test-http/src/test/java/org/elasticsearch/http/DanglingIndicesRestIT.java

```diff
@@ -0,0 +1,291 @@
+        // Ensure these parameters is accepted
+        deleteRequest.addParameter("timeout", "20s");
+        deleteRequest.addParameter("master_timeout", "20s");
+        final Response deleteResponse = restClient.performRequest(deleteRequest);
+        assertThat(deleteResponse.getStatusLine().getStatusCode(), equalTo(ACCEPTED.getStatus()));
+
+        assertBusy(() -> assertThat("Expected dangling index to be deleted", listDanglingIndexIds(), hasSize(0)));
+    }
+
+    private List<String> listDanglingIndexIds() throws IOException {
+        final Response response = getRestClient().performRequest(new Request("GET", "/_dangling"));
+        assertOK(response);
+
+        final XContentTestUtils.JsonMapView mapView = createJsonMapView(response.getEntity().getContent());
+
+        assertThat(mapView.get("_nodes.total"), equalTo(3));
+        assertThat(mapView.get("_nodes.successful"), equalTo(3));
+        assertThat(mapView.get("_nodes.failed"), equalTo(0));
+
+        List<Object> indices = mapView.get("dangling_indices");
+
+        List<String> danglingIndexIds = new ArrayList<>();
+
+        for (int i = 0; i < indices.size(); i++) {
+            danglingIndexIds.add(mapView.get("dangling_indices." + i + ".index_uuid"));
+        }
+
+        return danglingIndexIds;
+    }
+
+    private void assertOK(Response response) {
+        assertThat(response.getStatusLine().getStatusCode(), equalTo(OK.getStatus()));
+    }
+
+    /**
+     * Given a node name, finds the corresponding node ID.
+     */
+    private String mapNodeNameToId(String nodeName) throws IOException {
+        final Response catResponse = getRestClient().performRequest(new Request("GET", "/_cat/nodes?full_id&h=id,name"));
+        assertOK(catResponse);
+
+        for (String nodeLine : Streams.readAllLines(catResponse.getEntity().getContent())) {
+            String[] elements = nodeLine.split(" ");
+            if (elements[1].equals(nodeName)) {
+                return elements[0];
+            }
+        }
+
+        throw new AssertionError("Failed to map node name [" + nodeName + "] to node ID");
+    }
+
+    /**
+     * Helper that creates one or more indices, and importantly,
+     * checks that they are green before proceeding. This is important
+     * because the tests in this class stop and restart nodes, assuming
+     * that each index has a primary or replica shard on every node, and if
+     * a node is stopped prematurely, this assumption is broken.
+     *
+     * @return a mapping from each createad index name to its UUID
+     */
+    private Map<String, String> createIndices(String... indices) throws IOException {
+        assert indices.length > 0;
+        for (String index : indices) {
+            String indexSettings = "{"
+                + "  \"settings\": {"
+                + "    \"index\": {"
+                + "      \"number_of_shards\": 1,"
+                + "      \"number_of_replicas\": 2"
```

Comment:
I think we also need to set `index.routing.allocation.total_shards_per_node: 1` to ensure that every node gets a copy of the shard.